Library/Homebrew/keg_relocate.rb

```diff
@@ -165,6 +165,49 @@ def replace_text_in_files(relocation, files: nil)
     changed_files
   end
 
+  def relocate_build_prefix(keg, old_prefix, new_prefix)
+    # Find binaries which match prefix strings.
+    string_matches = Set.new
+    keg.each_unique_file_matching(old_prefix) do |file|
+      string_matches << file
+    end
+
+    binary_string_matches = Set.new
+    keg.each_unique_binary_file do |file|
+      binary_string_matches << file if string_matches.include?(file)
+    end
```

Comment:
Might need a refactoring of `each_unique_binary_file` but it looks like you could make this only read the `string_matches` files if you're only adding if matches are already in there.

Reply:
I tested this informally by calling `egrep -rlO "\x00" $(fgrep -rlO "/usr/local")` and it is noticeably faster.  I think all I need to do is add an optional `files` argument to `each_unique_binary_file` (similar to what we do for some other methods) which just passes the file list to `egrep` so it only scans those files instead of the whole keg.  I'll have to slightly tweak the unit test as well but it will be simple enough.

Reply:
This ended up being even simpler than I thought.  I just added `files` as an argument to `each_unique_binary_file` and initialized it to `nil`. If it's empty, `egrep` will just scan all files on the keg.  If it's not empty, only the files passed to it will be scanned, which is now what happens in `relocate_build_prefix`.  Testing this in some formulae with a large number of files, this does lead to a substantial speedup so this was a good improvement.

Reply:
Makes sense to me!